test-network/network.sh

```diff
@@ -267,9 +267,22 @@ function networkUp() {
 # call the script to create the channel, join the peers of org1 and org2,
 # and then update the anchor peers for each organization
 function createChannel() {
-  # Bring up the network if it is not already up.
+  # Bring up the network if it is not already up. 
+  bringUpNetwork="false"
 
-  if [ ! -d "organizations/peerOrganizations" ]; then
+  if ! $CONTAINER_CLI info > /dev/null 2>&1 ; then
+    fatalln "$CONTAINER_CLI network is required to be running to create a channel"
+  fi
+
+  # check if all containers are present 
+  CONTAINERS=($($CONTAINER_CLI ps | grep hyperledger/ | awk '{print $2}'))
+  len=$(echo ${#CONTAINERS[@]})
+  echo "len: $len"
+  [[ $len -ge 4 ]] && [[ ! -d "organizations/peerOrganizations" ]]  &&  networkDown 
+
```

Comment:
Its a nit, but you may want to display a message that you are bringing the network or any running containers down here if you are running the down command.

Reply:
Added in echo